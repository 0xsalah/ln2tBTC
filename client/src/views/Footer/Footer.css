footer {
    position: absolute;
    bottom: 0px;
    padding-bottom:10px;
}

footer > a{
    color:black;
    text-decoration: none;
    font-size: calc(8px + 1vmin);
}