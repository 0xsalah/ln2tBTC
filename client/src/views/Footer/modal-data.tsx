import React from "react";

interface modalDataType {
  title: string;
  buttonText: string;
  content: any;
}

const modalData: { [key: string]: modalDataType } = {
  code: {
    title: "Code",
    buttonText: "Got it!",
    content: <>Github!</>,
  },
  docs: {
    title: "Docs",
    buttonText: "Got it!",
    content: (
      <>
        <p>
          LN2tBTC is a decentralized service that enables trustless swaps
          between tBTC and BTC on the lightning network.
        </p>
        <p>
          It is based on a protocol derived from submarine swaps and it's served
          by a network of liquidity providers that anyone can join.
        </p>
        <p>
          If you'd like to learn more about how everything works check out our
          our docs.
        </p>
        <p>
          <a href="/">Learn more in this Medium artcile</a>
        </p>
      </>
    ),
  },
  about: {
    title: "About",
    buttonText: "Got it!",
    content: (
      <>
        LN2tBTC is a decentralized service that enables trustless swaps between
        tBTC and BTC on the lightning network.
        <br />
        <br />
        It is based on a protocol derived from submarine swaps and it's served
        by a network of liquidity providers that anyone can join.
        <br />
        <br />
        If you'd like to learn more about how everything works check out our{" "}
        <a href="https://github.com/corollari/ln2tBTC/blob/master/README.md">
          our docs
        </a>
        .
      </>
    ),
  },
  "contact-us": {
    title: "Contact us",
    buttonText: "Got it!",
    content: (
      <p>
        Just drop a message on Keep's discord tagging me{" "}
        <strong>@corollari#2127</strong> or send me an email at{" "}
        <a href="mailto:admin@ln2tbtc.com">admin@ln2tbtc.com</a>.
      </p>
    ),
  },
};

export default modalData;
