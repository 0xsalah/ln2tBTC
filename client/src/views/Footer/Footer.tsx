import React from "react";
import AlertDialog from "../AlertDialog/AlertDialog";
import "./Footer.css";

export default function Footer() {
  const [dialog, setDialog] = React.useState<"about" | "contact" | null>(null);

  const handleClose = () => {
    setDialog(null);
  };
  const a = <>a</>;
  return (
    <footer>
      <a href="https://github.com/corollari/ln2tBTC">Code</a>&nbsp;·&nbsp;
      <a href="https://github.com/corollari/ln2tBTC/blob/master/README.md">
        Docs
      </a>
      &nbsp;·&nbsp;
      <button
        type="button"
        className="link-button"
        onClick={() => setDialog("about")}
      >
        About
      </button>
      &nbsp;·&nbsp;
      <button
        type="button"
        className="link-button"
        onClick={() => setDialog("contact")}
      >
        Contact us
      </button>
      <AlertDialog
        title={dialog === "about" ? "About" : "Contact us"}
        open={dialog !== null}
        handleClose={handleClose}
      >
        {dialog === "about" ? (
          <>
            LN2tBTC is a decentralized service that enables trustless swaps
            between tBTC and BTC on the lightning network.
            <br />
            <br />
            It is based on a protocol derived from submarine swaps and it's
            served by a network of liquidity providers that anyone can join.
            <br />
            <br />
            If you'd like to learn more about how everything works check out our{" "}
            <a href="https://github.com/corollari/ln2tBTC/blob/master/README.md">
              our docs
            </a>
            .
          </>
        ) : (
          <>
            Just drop a message on{" "}
            <a href="https://discord.com/channels/590951101600235531/681226760209432608/748490345217654786">
              Keep's discord
            </a>{" "}
            tagging me (@corollari#2127) or send me an email at
            admin@ln2tbtc.com
          </>
        )}
      </AlertDialog>
    </footer>
  );
}
