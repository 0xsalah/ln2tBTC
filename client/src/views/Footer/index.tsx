import React from "react";
import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import DialogTitle from "@material-ui/core/DialogTitle";
import Slide from "@material-ui/core/Slide";
import { TransitionProps } from "@material-ui/core/transitions";
import "./Footer.css";

const Transition = React.forwardRef(function Transition(
  props: TransitionProps & { children?: React.ReactElement<any, any> },
  ref: React.Ref<unknown>
) {
  return <Slide direction="up" ref={ref} {...props} />;
});

export default function AlertDialog() {
  const [dialog, setDialog] = React.useState<"about" | "contact" | null>(null);

  const handleClose = () => {
    setDialog(null);
  };

  return (
    <footer>
      <a href="https://github.com/corollari/ln2tBTC">Code</a>&nbsp;·&nbsp;
      <a href="https://github.com/corollari/ln2tBTC/blob/master/README.md">
        Docs
      </a>
      &nbsp;·&nbsp;
      <a onClick={() => setDialog("about")} href="#">
        About
      </a>
      &nbsp;·&nbsp;
      <a onClick={() => setDialog("contact")} href="#">
        Contact us
      </a>
      <Dialog
        open={dialog !== null}
        TransitionComponent={Transition}
        keepMounted
        onClose={handleClose}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description"
      >
        <DialogTitle id="alert-dialog-title">
          {dialog === "about" ? "About" : "Contact us"}
        </DialogTitle>
        <DialogContent>
          <DialogContentText id="alert-dialog-description">
            {dialog === "about" ? (
              <>
                LN2tBTC is a decentralized service that enables trustless swaps
                between tBTC and BTC on the lightning network.
                <br />
                <br />
                It is based on a protocol derived from submarine swaps and it's
                served by a network of liquidity providers that anyone can join.
                <br />
                <br />
                If you'd like to learn more about how everything works check out
                our{" "}
                <a href="https://github.com/corollari/ln2tBTC/blob/master/README.md">
                  our docs
                </a>
                .
              </>
            ) : (
              <>
                Just drop a message on{" "}
                <a href="https://discord.com/channels/590951101600235531/681226760209432608/748490345217654786">
                  Keep's discord
                </a>{" "}
                tagging me (@corollari#2127) or send me an email at
                admin@ln2tbtc.com
              </>
            )}
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose} color="primary" autoFocus>
            Close
          </Button>
        </DialogActions>
      </Dialog>
    </footer>
  );
}
/*
export default function(){
    return <footer><a href="https://github.com/corollari/ln2tBTC">Code</a> ·
    <a href="https://github.com/corollari/ln2tBTC/blob/master/README.md">Docs</a> ·
    <a>Chat</a></footer>
}
*/
