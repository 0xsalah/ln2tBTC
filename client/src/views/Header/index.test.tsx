import React from "react";
import { render } from "@testing-library/react";
import Footer from ".";

test("snapshot of component rendered without web3", () => {
  const component = render(<Footer />);
  expect(component.baseElement).toMatchSnapshot();
});
