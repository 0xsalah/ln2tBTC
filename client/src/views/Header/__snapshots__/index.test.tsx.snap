// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of component rendered without web3 1`] = `
<body>
  <div>
    <header
      class="header header--short"
    >
      <div
        class="container"
      >
        <div
          class="header__row row justify-between"
        >
          <div
            class="header__left"
          >
            <a
              class="logo"
              href="/"
            >
              <img
                alt=""
                class="logo__img"
                height="66"
                src="logo.svg"
                width="111"
              />
            </a>
          </div>
          <div
            class="header__right row"
          >
            <div
              class="header__connect connect connect--no"
            >
              <div
                class="connect__label"
              >
                Connect Wallet
              </div>
              <div
                class="connect__status"
              >
                <span />
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>
</body>
`;
